django/db/views.py

```diff
@@ -2111,6 +2123,16 @@ def small_molecules_for_dataset(self,dataset_id):
 #        # view of the small molecule table
 #        logger.info(str(('small molecules for dataset',dataset_id,len(queryset))))
 #        return queryset
+
+    # Need a second distinction, because we'll only show the sm column 
+    # if there are datarecord entries with cells
+    def has_small_molecules_for_datarecords(self, dataset_id):
```

Comment:
See above question on whether dataset_id should be an arg here, or just retrieved via self.